src/main/java/com/shipmentEvents/handlers/EventHandler.java

```diff
@@ -97,51 +95,70 @@ private void processShipmentUpdates(final LambdaLogger logger) throws Interrupte
         final AmazonS3 s3Client = EventHandler.getS3Client();
         logger.log("Processing Bucket: " + bucketName);
 
-        ObjectListing files = s3Client.listObjects(bucketName);
+        String continuationToken = null;
+        boolean keepReading = true;
+
         List<KeyVersion> filesProcessed = new ArrayList<DeleteObjectsRequest.KeyVersion>();
 
-        for (Iterator<?> iterator = files.getObjectSummaries().iterator(); iterator.hasNext(); ) {
-            S3ObjectSummary summary = (S3ObjectSummary) iterator.next();
-            logger.log("Reading Object: " + summary.getKey());
+        while (keepReading) {
+            ListObjectsV2Request request = new ListObjectsV2Request();
+            request.setBucketName(bucketName);
+            if (continuationToken != null) {
+                request.setContinuationToken(continuationToken);
+            }
+            ListObjectsV2Result files = s3Client.listObjectsV2(request);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code is manually paginating results. You could use the SDK to automatically paginate these results instead. SDK operation name: [S3Objects.withPrefix](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/iterable/S3Objects.html#withPrefix-com.amazonaws.services.s3.AmazonS3-java.lang.String-java.lang.String-).
 The list of [S3ObjectSummary](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/model/S3ObjectSummary.html)s will be fetched lazily, a page at a time, as they are needed. The size of the page can be controlled with the [withBatchSize(int)](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/iterable/S3Objects.html#withBatchSize-int-) method.